### ISSUE TITLE
ndarray.split function not working as instruction

### ISSUE THREAD

**Issue by Godricly** [dmlc/mxnet#5790]
For bugs or installation issues, please provide the following information.
The more information you provide, the more likely people will be able to help you.

## Environment info
Operating System:
Linux
Compiler:
G++
Package used (Python/R/Scala/Julia):
Python
MXNet version:

Or if installed from source:

MXNet commit hash (`git rev-parse HEAD`):
22482aecf99ed6e1bc82803845aa53dbddb835dc
If you are using python package, please provide

Python version and distribution:
2.7
If you are using R package, please provide

R `sessionInfo()`:

## Error Message:
Please paste the full error message, including stack trace.
***
Traceback (most recent call last):
  File "test.py", line 9, in <module>
    splits= mx.nd.split(a,axis=1,num_outputs=4)
  File "mxnet_path/python/mxnet/_ctypes/ndarray.py", line 164, in generic_ndarray_function
    c_array(ctypes.c_char_p, [c_str(val) for val in vals])))
  File "mxnet_path/python/mxnet/base.py", line 78, in check_call
    raise MXNetError(py_str(_LIB.MXGetLastError()))
mxnet.base.MXNetError: Invalid Parameter format for num_outputs expect int but value='<NDArray 5x4x2x2 @cpu(0)>', in operator SliceChannel(name="", axis="1", num_outputs="<NDArray 5x4x2x2 @cpu(0)>")
***
## Minimum reproducible example
if you are using your own code, please provide a short script that reproduces the error.
```python
import mxnet as mx

a = mx.nd.zeros((5,4,2,2))
splits= mx.nd.split(a,axis=1,num_outputs=4)
print splits
```
## Steps to reproduce
or if you are running standard examples, please provide the commands you have run that lead to the error.

1. following example
```python
x = [[1, 2],
     [3, 4],
     [5, 6],
     [7, 8]]  // 4x2 array

y = split(x, axis=0, num_outputs=4) // a list of 4 arrays
y[0] = [[ 1.,  2.]]  // 1x2 array

z = split(x, axis=0, num_outputs=2) // a list of 2 arrays
z[0] = [[ 1.,  2.],
        [ 3.,  4.]]
```

## others
1. In the mx.nd.Concat  function, the C in upper case. While the mx.nd.split, the s is in lower case. The naming should be unified.
 @tqchen @piiswrong

**Comment by Godricly** [comment]
It seems that the ndarry to split is not passed in correctly.

**Comment by Godricly** [comment]
fixed in #5791.
